gtsfm/utils/images.py

```diff
@@ -118,6 +118,45 @@ def resize_image(image: Image, new_height: int, new_width: int) -> Image:
     return Image(resized_value_array)
 
 
+def get_downsample_factor_per_axis(img: Image, max_resolution: int) -> Tuple[float, float, int, int]:
+    """
+    Resizing an image by a specific downsample ratio may not be possible due to lack of a clean
+    divisor. However, we can still determine the exact downsampling ratio.
+
+    Args:
+        img: image to be downsampled
+        max_resolution: integer representing maximum length of image's short side
+            e.g. for 1080p (1920 x 1080), max_resolution would be 1080
+
+    Returns:
+        downsample_u: downsampling factor for u coordinate
+        downsample_v: downsampling factor for v coordinate
+        new_h: new height that will preserve aspect ratio as closely as possible, while
+            respecting max_resolution constraint
+        new_w: new width
+    """
+    h, w = img.height, img.width
+    shorter_size = min(h, w)
+    if shorter_size == h:
+        new_h = max_resolution
+        # compute scaling that will be applied to original image
+        scale = new_h / float(h)
+        new_w = int(w * scale)
+    else:
+        new_w = max_resolution
+        scale = new_w / float(w)
+        new_h = int(h * scale)
```

Comment:
This would work even if max_resolution is greater than the input image dimensions right? Can we add a check to make sure it is not? I dont think it is expected to work, since the function mentions "downsampling".